pynvim/msgpack_rpc/event_loop/asyncio.py

```diff
@@ -188,18 +193,33 @@ async def connect_stdout():
 
     @override
     def _connect_child(self, argv: List[str]) -> None:
-        if os.name != 'nt':
-            # see #238, #241
-            self._child_watcher = asyncio.get_child_watcher()
-            self._child_watcher.attach_loop(self._loop)
-
         async def create_subprocess():
             transport: asyncio.SubprocessTransport  # type: ignore
             transport, protocol = await self._loop.subprocess_exec(
                 self._protocol_factory, *argv)
             pid = transport.get_pid()
             debug("child subprocess_exec successful, PID = %s", pid)
 
+            if os.name != 'nt':
+                # see #238, #241
+                pidfd_works = False
+                if PidfdChildWatcher is not None and hasattr(os, "pidfd_open"):
+                    try:
+                        fd = os.pidfd_open(pid)
+                    except Exception:
```

Comment:
Out of curiosity, under which settings can the `pidfd_open` syscall fail?

Reply:
It is a Linux syscall. So on Linux < 5.3 or BSD.

Reply:
On Linux < 5.3, `os.pidfd_open` won't exist at all. Precisely what exceptions are being caught, and when does it *fail* (while exists)?

Reply:
Invalid pid, pid already dead, not sure if a pid is a zombie. Are we sure transport.get_pid() only returned a valid working pid?